main.go

```diff
@@ -226,12 +230,29 @@ func main() {
 	if *dataPath != "" {
 		dir = *dataPath
 	} else {
-		dir, err = DetectPath(os.Getenv("GOPATH"))
+		goPath, isExist := os.LookupEnv("GOPATH")
+		if isExist {
+			fmt.Println("$GOPATH has been set explicitly:", goPath)
+			fmt.Printf("Searching directory '%s' in '%s'...\n", defaultDataPath, goPath)
+		} else {
+			output, err := exec.Command("go", "env", "GOPATH").Output()
+			if err != nil {
+				fmt.Println("Failed: please set '$GOPATH' manually, or use 'datapath' option to specify the path to your custom 'data' directory")
+				return
+			}
+			goPath = string(output)
+			goPath = strings.TrimSuffix(goPath, "\n")
+			goPath = strings.TrimSuffix(goPath, "\r")
+			fmt.Println("Use default $GOPATH set by golang:", goPath)
+			fmt.Printf("Searching directory '%s' in '%s'...\n", defaultDataPath, goPath)
+		}
```

Comment:
They can be replaced by 
```go
build.Default.GOPATH // import "go/build"
```

Reference:
https://golang.org/src/go/build/build.go#L273 => https://golang.org/src/go/build/build.go#L296

Reply:
I have tested it.

If you do not set `$GOPATH` environment variable explicitly, then Go runtime will use the one in `go env GOPATH` by default. In this case, if you have set `go env -w GOPATH=/a/b/c`, then `go get -u github.com/a/repo` will be installed in `/a/b/c/src/github.com/a/repo`, not the one of `build.Default.GOPATH`.

Then the build process will fail.

Reply:
![image](https://user-images.githubusercontent.com/10487845/87773546-e7812f80-c855-11ea-9d82-ae70be2dbcb7.png)

Reply:
It seems that `go env -w` will write environment into **EnvFile**.
So how about reading file from **EnvFile**, which can be found by a function like this https://github.com/golang/go/blob/c4f2a9788a7be04daf931ac54382fbe2cb754938/src/cmd/go/internal/cfg/cfg.go#L151 ?

I don't recommend using `exec.Command`, cause in some case, you may face some complex outside-environment problems.

Reply:
It seems that's an internal package.

Reply:
yes, I mean build a function like this